### PR TITLE
Hyperledger Fabric 2.x chaincode container

### PR DESCRIPTION
**Pull request by Kelvin-M**
As Hyperledger Fabric 2.x exposed a new chaincode lifecycle, current chaincode needs update : 
- Dockerfile to build its own image
- Expose chaincode as a server
- Update dependencies

Companion PR : 
https://github.com/SubstraFoundation/substra-backend/pull/343
https://github.com/SubstraFoundation/hlf-k8s/pull/97

Reminder : remove build from docker-hub (branch hlf-2) @AlexandrePicosson

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# Build application
+RUN go build -o chaincode -v .
+
+# Production ready image
+# Pass the binary to the prod image
+FROM alpine:3.11 as prod
+
+COPY --from=build /go/src/github.com/chaincode/chaincode /app/chaincode
+
+USER 1000
```
Comment (AurelienGasser) [inline Dockerfile:15]: :ok_hand:

**Review by AurelienGasser — APPROVED** [review]
Nice work!

File: chaincode/main.go [hunk 2]
```diff
@@ -226,11 +228,45 @@ func formatErrorResponse(err error) peer.Response {
...
-	// TODO use the same level as the shim or an env variable
-	logger.SetLevel(shim.LogDebug)
-	if err := shim.Start(new(SubstraChaincode)); err != nil {
-		fmt.Printf("Error starting SubstraChaincode chaincode: %s", err)
+
+	logger.SetOutput(os.Stdout)
+	logger.SetLevel(logrus.DebugLevel)
+	logger.Infof("Load TLS certificates")
+
+	key, err := ioutil.ReadFile("/var/hyperledger/tls/client/pair/tls.key")
```
Comment (inelgnu) [inline chaincode/main.go]: maybe add ENV variables to pass certificates, tls keys, etc... here and below hardcoded values could lead to some limitations running the chaincode locally or in a different context

Comment (Kelvin-M) [reply chaincode/main.go]: Yes, I will do the update 👍

Comment (Kelvin-M) [reply chaincode/main.go]: Done !

**Review by inelgnu — APPROVED** [review]
LGTM

**Pull request merged**